bin/p2-exec/main.go

```diff
@@ -37,6 +44,14 @@ func main() {
 	kingpin.Version(version.VERSION)
 	kingpin.Parse()
 
+	if *umask != umaskSentinal {
+		effectiveUmask, err := strconv.ParseInt(*umask, 8, 0)
+		if err != nil {
+			log.Fatalf("umask not expressed in octal notation. %v\n", err)
+		}
+		unix.Umask(int(effectiveUmask))
```

Comment:
This function does not return an error. The return value is the old mask which is not interesting.